contracts/governance/src/ZeroExVotes.sol

```diff
@@ -0,0 +1,232 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+
+  Copyright 2023 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+pragma solidity ^0.8.17;
+
+import "@openzeppelin/utils/math/SafeCast.sol";
+import "@openzeppelin/utils/math/Math.sol";
+import "@openzeppelin/token/ERC20/ERC20.sol";
+import "@openzeppelin/governance/utils/IVotes.sol";
+import "./IZeroExVotes.sol";
+
+/// @custom:security-contact security@0xproject.com
+contract ZeroExVotes is IZeroExVotes {
+    address public token;
```

Comment:
Had to revert this as the `token` is now set in the `initialise` function rather than the constructor.

Reply:
Why is `token` set in the initializer? That makes all operations ~2.1k gas more expensive with no benefit. Is it because the OZ upgrades utility throws a fit when proxy implementations have constructors?

Reply:
That but also because we have a catch 22 with the token using the voting implementation in its constructor and voting needing the token as well.

Reply:
Because neither the token wrapper nor the votes contract make any calls each other as part of the constructor, it's not a big deal to take each others' addresses as arguments. Knowing the deployer address and the future deployment nonce, we can precompute the address that the votes contract will deploy to and pass that as the argument to the wrapper (or vice versa). Then, later, deploy the votes contract to that address.

This _would_ be a circular dependency if we were intending to deploy these contracts with CREATE2, but I don't think we're intending to do that.

Reply:
I've implemented OZ's initializable for the voting contract, see if you feel better about it

Reply:
This is more of a gas thing than missing `Initializeable`. If `token` is immutable, we get to avoid a cold `SLOAD` on each `onlyToken` call.

Reply:
This is more of a gas thing than missing `Initializeable`. If `token` is immutable, we get to avoid a cold `SLOAD` on each `onlyToken` call.

Reply:
There is still a circular dependency though. 
voting proxy requires voting implementation
voting implementation requires wrapped token
wrapped token requires voting proxy
🤷‍♀️

Reply:
But they _do_ require each other's addresses in their constructor / initializer logic 😕

Reply:
`ERC1967Proxy` does check that code is present at the implementation address. This happens in its constructor.

Reply:
Good catch. So we can't choose to deploy the votes implementation last. Deploying the votes proxy last doesn't help us at all. We _can_ choose to deploy wrapped token last, though.

Reply:
Sorry I don't understand. Happy for you to push any updates you want for this.

Reply:
I made #680 . If you merge that, it implements the changes I suggested above.